app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -209,33 +223,42 @@ class UndertakingController @Inject()(
 
   def postCheckAnswers: Action[AnyContent] = escAuthentication.async { implicit request =>
     implicit val eori: EORI = request.eoriNumber
+
     cyaForm.bindFromRequest().fold(
-      _ => throw new IllegalStateException("value hard-coded, form hacking?"),
+      _ => sys.error("value hard-coded, form hacking?"),
```

Comment:
`sys.error` just returns a RuntimeException. Might it be better to retain the meaning of the `IllegalStateException` here?

Reply:
Throughout the code , only this exception is being used "throw new IllegalStateException", even if it's not required. Here it's checking if the form has some value or not. was not sure if that's part of IllegalStateException. what do you think?

Reply:
Can we do something better here than throwing an exception?

Reply:
I think it should throw exception here as the value is needed. Although in real life scenario, it will never happen but theoretically it should throw an exception

Reply:
Ok, maybe leave as is for now. We can always revisit later.